Jenkinsfile

```diff
@@ -1,354 +1,5 @@
-            }
-            lib_West.InitUpdate('zephyr')
-            lib_West.ApplyManifestUpdates(CI_STATE)
-            dir('zephyr') {
-              sh """
-                  export ZEPHYR_TOOLCHAIN_VARIANT='$compiler' && \
-                  source zephyr-env.sh && \
-                  ./scripts/sanitycheck $SANITYCHECK_OPTIONS $ARCH --subset $subset || $SANITYCHECK_RETRY_CMDS
-                 """
-            }
-          }
-        }
-      } finally {
-        cleanWs(); echo "Ran: cleanWs()"
-      }
-    }
-  }
-}
-
-def generateComplianceStage(AGENT_LABELS, DOCKER_REG, IMAGE_TAG, JOB_NAME, CI_STATE) {
-  return {
-    node (AGENT_LABELS) {
-      try {
-        stage('Compliance Test'){
-          println "Using Node:$NODE_NAME"
-          docker.image("$DOCKER_REG/$IMAGE_TAG").inside {
-            dir('zephyr') {
-              checkout scm
-              CI_STATE.ZEPHYR.REPORT_SHA = lib_Main.checkoutRepo(
-                    CI_STATE.ZEPHYR.GIT_URL, "ZEPHYR", CI_STATE.ZEPHYR, false)
-              lib_Status.set("PENDING", 'ZEPHYR', CI_STATE);
-              lib_West.AddManifestUpdate("ZEPHYR", 'zephyr',
-                    CI_STATE.ZEPHYR.GIT_URL, CI_STATE.ZEPHYR.GIT_REF, CI_STATE)
-            }
-            lib_West.InitUpdate('zephyr')
-            lib_West.ApplyManifestUpdates(CI_STATE)
-
-            dir('zephyr') {
-              def BUILD_TYPE = lib_Main.getBuildType(CI_STATE.ZEPHYR)
-              if (BUILD_TYPE == "PR") {
-                if (CI_STATE.ZEPHYR.IS_MERGEUP) {
-                  println 'This is a MERGE-UP PR.   CI_STATE.ZEPHYR.IS_MERGEUP=' + CI_STATE.ZEPHYR.IS_MERGEUP
-                  CI_STATE.ZEPHYR.MERGEUP_BASE = sh( script: "git log --oneline --grep='\\[nrf mergeup\\].*' -i -n 1 --pretty=format:'%h' | tr -d '\\n'" , returnStdout: true)
-                  println "CI_STATE.ZEPHYR.MERGEUP_BASE = $CI_STATE.ZEPHYR.MERGEUP_BASE"
-                  COMMIT_RANGE = "$CI_STATE.ZEPHYR.MERGEUP_BASE..$CI_STATE.ZEPHYR.REPORT_SHA"
-                } else {
-                  COMMIT_RANGE = "$CI_STATE.ZEPHYR.MERGE_BASE..$CI_STATE.ZEPHYR.REPORT_SHA"
-                }
-                COMPLIANCE_ARGS = "$COMPLIANCE_ARGS -p $CHANGE_ID -S $CI_STATE.ZEPHYR.REPORT_SHA -g"
-                println "Building a PR [$CHANGE_ID]: $COMMIT_RANGE"
-              }
-              else if (BUILD_TYPE == "TAG") {
-                COMMIT_RANGE = "tags/${env.BRANCH_NAME}..tags/${env.BRANCH_NAME}"
-                println "Building a Tag: " + COMMIT_RANGE
-              }
-              // If not a PR, it's a non-PR-branch or master build. Compare against the origin.
-              else if (BUILD_TYPE == "BRANCH") {
-                COMMIT_RANGE = "origin/${env.BRANCH_NAME}..HEAD"
-                println "Building a Branch: " + COMMIT_RANGE
-              }
-              else {
-                  assert condition : "Build fails because it is not a PR/Tag/Branch"
-              }
+def pipeline = new ncs.sdk_zephyr.Main()
```

Comment:
Hard to evaluate what happens here, so I guess I just have to trust that code will do what we want.

But regarding this PR itself, I was wondering if it would be better to make a revert of this PR:
https://github.com/nrfconnect/sdk-zephyr/commit/b52cb0e67fac97823a53766813692fdee211aea0

and then re-add the ` .github/workflows/backport.yml` file to this PR.
that should minimize the patchset during upmerge.
@mbolivar-nordic thoughts ?

Reply:
I agree that would be cleaner, but I don't think it's a blocker. I'll notice in time for v1.4.0 and rewrite the history to make it so if this suggestion is not applied during this PR.

Reply:
I am on leave for next 3 months.  

@asle-nordic can you follow-up on this PR?